src/main/java/core/basesyntax/service/impl/DataConverterImpl.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.DataConverter;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> inputReport) {
+        List<FruitTransaction> fruitTransactionList = new ArrayList<>();
+        for (String item: inputReport) {
+            String[] arrayFromItem = item.replaceAll(" ", "").split(",");
```

Comment:
Consider adding validation to ensure that `arrayFromItem` has the expected number of elements before accessing them. This will prevent potential `ArrayIndexOutOfBoundsException` if the input data is malformed.